src/protocols/XDGShell.cpp

```diff
@@ -534,30 +534,36 @@ CBox CXDGPositionerRules::getPosition(const CBox& constraint, const Vector2D& pa
     if (success)
         return predictedBox.translate(-parentCoord - constraint.pos());
 
+    CBox test = predictedBox;
+
     if (state.constraintAdjustment & (XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_X | XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_Y)) {
         // attempt to flip
-        const bool flipX = state.constraintAdjustment & XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_X;
-        const bool flipY = state.constraintAdjustment & XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_Y;
-
-        CBox       test = predictedBox;
-        success         = true;
-        if (flipX && test.copy().translate(Vector2D{-predictedBox.w - state.anchorRect.w, 0}).expand(-1).inside(constraint))
+        const bool flipX      = state.constraintAdjustment & XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_X;
+        const bool flipY      = state.constraintAdjustment & XDG_POSITIONER_CONSTRAINT_ADJUSTMENT_FLIP_Y;
+        auto       countEdges = [constraint](const CBox& test) -> int {
+            int edgeCount = 0;
+            edgeCount += test.x < constraint.x ? 1 : 0;
+            edgeCount += test.x + test.w > constraint.x + constraint.w ? 1 : 0;
+            edgeCount += test.y < constraint.y ? 1 : 0;
+            edgeCount += test.y + test.h > constraint.y + constraint.h ? 1 : 0;
+            return edgeCount;
+        };
+        int edgeCount = countEdges(test);
+
+        // -1 as we want 2 edges to be resolved with the flip, otherwise move on to only x or only y
+        if (flipX && flipY && edgeCount - 1 > countEdges(test.copy().translate(Vector2D{-predictedBox.w - state.anchorRect.w, -predictedBox.h - state.anchorRect.h})))
```

Comment:
this case makes no sense. You can drop this one and instead use the two ifs below without else

Reply:
Ah yeah, thanks.